polaris/utils/types.py

```diff
@@ -128,7 +138,7 @@ class License(BaseModel):
         "https://raw.githubusercontent.com/spdx/license-list-data/main/json/licenses.json"
     )
 
-    id: str
+    id: SupportedLicenseType | str
     reference: Optional[HttpUrlString] = None
```

Comment:
We should automatically compute the `reference` attribute based on the `id` if the `id` is not `None`. Because we don't want the user to be able to specify a custom link anymore, we should turn the `reference` into a [Pydantic `@computed_field`](https://docs.pydantic.dev/2.0/usage/computed_fields/).